### PR TITLE
Create a container that holds hypermediaState obj

### PR DESCRIPTION
**Pull request by ciss1995**

### CODE REVIEW

File: state/store.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import SirenParse from 'siren-parser';
 
 class StateStore {
 	constructor(fetch) {
+		this.states = new Map();
```
Comment (tenbradley) [inline state/store.js:9]: Is the intent to have both this.states and this._states?

File: state/store.js [hunk 2]
```diff
@@ -264,3 +269,13 @@ export async function performAction(action, input) {
 export async function dispose(state, component) {
 	state && state.dispose(component);
 }
+
+function _addStateToMap(href, token, state) {
```
Comment (m6jones) [inline state/store.js:273]: This is probably more personal preference. 

I would rather the `token` object get passed here and we use the `token.toString()` in this method.

When we come back to it, that will make all the `token` variables represent the same thing.

File: state/store.js (hunk 1 above)
Comment (ciss1995) [reply state/store.js:9]: Humm good question. Both are doing a similar thing but slightly different. The `_states` stores the state as a key and count as val and state stores the state as value. We could definitely just combine the two if that's the direction we want to take

File: state/store.js (hunk 2 above)
Comment (ciss1995) [reply state/store.js:273]: Yeah, definitely something we can do

File: state/store.js (hunk 1 above)
Comment (tenbradley) [reply state/store.js:9]: We can have both. I would just suggest less similar names.

Comment (ciss1995) [reply state/store.js:9]: Good point, I'll rename the new one.

File: state/store.js [hunk 3]
```diff
@@ -6,6 +6,7 @@ import SirenParse from 'siren-parser';
 
 class StateStore {
 	constructor(fetch) {
+		this.cachedStates = new Map();
```
Comment (m6jones) [inline state/store.js:9]: This should be private?

File: state/store.js [hunk 4]
```diff
@@ -221,23 +222,27 @@ export async function refreshState(state, refetch = true) {
 	return window.D2L.SirenSdk.StateStore.fetch(state, refetch);
 }
 
-export function stateFactoryByRawSirenEntity(rawEntity, token) {
-	const entityId = getEntityIdFromSirenEntity(rawEntity);
+export async function stateFactory(entity, token) {
+	if (!entity) return;
+
+	const entityId = entity.constructor.name === 'Link' ? getEntityIdFromSirenEntity(entity) : entity;
```
Comment (m6jones) [inline state/store.js:228]: I would do this this other way. Where I'd check to see if entity is a string. If it isn't run it through `getEntityIdFromSirenEntity`

**Comment by m6jones** [comment]
I updated the notes in the story:
https://rally1.rallydev.com/#/357252966636d/custom/367300408400?detail=%2Fuserstory%2F441183098312

I am going to split this into two stories. This is my bad :( 

Discussed with Ten and Lloyd on the call.

**Comment by ciss1995** [comment]
This should be replaced by https://github.com/BrightspaceHypermediaComponents/foundation-engine/pull/19 and https://github.com/BrightspaceHypermediaComponents/foundation-engine/pull/18.

**Pull request closed without merging**